### PR TITLE
[cli] cleanup CoAP example

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: src/cli/README_COAP.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+coap request from [fdde:ad00:beef:0:b3:e3f6:2dcc:4b79] GET
+coap response sent
+coap request from [fdde:ad00:beef:0:b3:e3f6:2dcc:4b79] PUT with payload: 7061796c6f6164
+coap response sent
+```
+
+## Command List
+
+* [help](#help)
+* [delete](#delete)
```
Comment (bukepo) [inline src/cli/README_COAP.md]: [delete](#delete-address-uri-path-type-payload)

File: src/cli/README_COAP.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+```bash
+> coap get fdde:ad00:beef:0:2780:9423:166c:1aac test-resource
+Done
+```
+
+### post \<address\> \<uri-path\> \[type\] \[payload\]
+
+* address: IPv6 address of the CoAP server.
+* uri-path: URI path of the resource.
+* type: "con" for Confirmable or "Non-con" for non-confirmable (default).
```
Comment (bukepo) [inline src/cli/README_COAP.md]: `Confirmable` -> `confirmable`?

File: src/cli/README_COAP.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+```bash
+> coap post fdde:ad00:beef:0:2780:9423:166c:1aac test-resource con payload
+Done
+```
+
+### put \<address\> \<uri-path\> \[type\] \[payload\]
+
+* address: IPv6 address of the CoAP server.
+* uri-path: URI path of the resource.
+* type: "con" for Confirmable or "Non-con" for non-confirmable (default).
```
Comment (bukepo) [inline src/cli/README_COAP.md]: ditto

File: src/cli/cli_coap.cpp [hunk 4]
```diff
@@ -323,26 +247,144 @@ otError Coap::ProcessRequest(int argc, char *argv[])
...
+        otCoapMessageInit(responseMessage, OT_COAP_TYPE_ACKNOWLEDGMENT, responseCode);
+        otCoapMessageSetMessageId(responseMessage, otCoapMessageGetMessageId(aMessage));
+        otCoapMessageSetToken(responseMessage, otCoapMessageGetToken(aMessage), otCoapMessageGetTokenLength(aMessage));
+
+        if (otCoapMessageGetCode(aMessage) == OT_COAP_CODE_GET)
+        {
+            otCoapMessageSetPayloadMarker(responseMessage);
+        }
+
+        if (otCoapMessageGetCode(aMessage) == OT_COAP_CODE_GET)
```
Comment (bukepo) [inline src/cli/cli_coap.cpp]: merge with above if?

File: src/cli/README_COAP.md (hunk 2 above)
Comment (jwhui) [reply src/cli/README_COAP.md]: Prefer capitalized terms to be consistent with [RFC 7252](https://tools.ietf.org/html/rfc7252).

File: src/cli/README_COAP.md (hunk 3 above)
Comment (jwhui) [reply src/cli/README_COAP.md]: See response above.

File: src/cli/README_COAP.md (hunk 2 above)
Comment (bukepo) [reply src/cli/README_COAP.md]: I just found `Confirmable` is not consistent with `non-confirmable`.

**Comment by jwhui** [comment]
@bukepo, can you point out where you see the inconsistencies in the prior push?

**Comment by bukepo** [comment]
@jwhui Just found they are updated. It seems I re-checked in wrong version or not got the page fully refreshed. Thanks 👍

**Pull request merged**